WIP/calculator.rb

```diff
@@ -0,0 +1,95 @@
+class Calculate
+
+  def initialize(**args)
+    @accumulate = args[:accumulate] || args[:accumulate] == nil
+    @discounts = {}
+  end
+
+  def add_discount(**args)
+
+    name = args[:name]
+    amount = args[:amount]
+    percentage = args[:percentage]
+
+
+    if amount.is_a?(String) || percentage.is_a?(String) || amount.to_i < 0  
+      raise StandardError.new "Not a valid deal"
+    elsif amount == nil && percentage == nil
+      raise StandardError.new "Not a valid deal"
+    elsif percentage.to_i > 100 || percentage.to_i < 0
+      raise StandardError.new "Not a valid deal"
+    end
+
+    @discounts[name] = {
+      name: name,
+      amount: amount,
+      percentage: percentage
+    }
+
+    true
+  end
+
+  def return_discounts
+    @discounts
+  end
+
+  def accumulate?
+    @accumulate
+  end
+
+  def remove_discount!(discount_name)
+    @discounts.delete(discount_name)
+  end
+
+  
+  def calculate_discount_accumulate_true(fullcost)
```

Comment:
did not review this. we will come back to this method once we have figured out accumulate false first.